internal/pkg/manifest/env.go

```diff
@@ -417,6 +418,66 @@ func (cfg *EnvironmentHTTPConfig) loadLBConfig(env *config.CustomizeEnv) {
 type PublicHTTPConfig struct {
 	SecurityGroupConfig ALBSecurityGroupsConfig `yaml:"security_groups,omitempty"`
 	Certificates        []string                `yaml:"certificates,omitempty"`
+	ELBAccessLogs       ELBAccessLogsArgsOrBool `yaml:"access_logs,omitempty"`
+}
+
+// ELBAccessLogsArgsOrBool is a custom type which supports unmarshaling yaml which
+// can either be of type bool or type ELBAccessLogsArgs.
+type ELBAccessLogsArgsOrBool struct {
+	Enabled        *bool
+	AdvancedConfig ELBAccessLogsArgs
+}
+
+func (al *ELBAccessLogsArgsOrBool) isEmpty() bool {
+	return al.Enabled == nil && al.AdvancedConfig.isEmpty()
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the ELBAccessLogsArgsOrBool
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (al *ELBAccessLogsArgsOrBool) UnmarshalYAML(value *yaml.Node) error {
+	if err := value.Decode(&al.AdvancedConfig); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !al.AdvancedConfig.isEmpty() {
+		// Unmarshaled successfully to al.AccessLogsArgs, reset al.EnableAccessLogs, and return.
+		al.Enabled = nil
+		return nil
+	}
+
+	if err := value.Decode(&al.Enabled); err != nil {
+		return errUnmarshalELBAccessLogs
+	}
+	return nil
+}
+
+// AccessLogsConfig holds the access logs configuration.
+type ELBAccessLogsArgs struct {
+	BucketName   *string `yaml:"bucket_name,omitempty"`
+	BucketPrefix *string `yaml:"bucket_prefix,omitempty"`
+}
+
+func (al *ELBAccessLogsArgs) isEmpty() bool {
+	return al.BucketName == nil && al.BucketPrefix == nil
+}
+
+// AccessLogs returns the access logs config if the user has set any values.
+// If there is no access logs settings, then returns nil and false.
+func (cfg *EnvironmentConfig) ELBAccessLogs() (*ELBAccessLogsArgs, bool) {
+	accessLogs := cfg.HTTPConfig.Public.ELBAccessLogs
+	if accessLogs.isEmpty() {
+		return nil, false
+	}
+	if accessLogs.Enabled != nil {
+		return nil, true
+	}
+	return &accessLogs.AdvancedConfig, true
```

Comment:
What about
```yaml
http:
  public:
    access_logs: false
```

Reply:
That's a great catch. I had thought that if they want to remove the access logs then they will just remove the configuration instead of setting this flag as false. But I have changed the logic to catch false and remove the access logs feature.